src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -195,7 +195,15 @@ const std::shared_ptr<const Table>& SQLPipelineStatement::get_result_table() {
   CurrentScheduler::schedule_and_wait_for_tasks(tasks);
 
   if (_auto_commit) {
-    _transaction_context->commit();
+    const auto success = _transaction_context->commit();
+
+    if (!success) {
+      _query_has_output = false;
+      return _result_table = nullptr;
```

Comment:
Rather do only one thing per line. Also, readers who have not seen our discussion will have a hard time understanding why this is done.